### PR TITLE
refactor(ckb-resource): Remove dependency on tempfile in ckb-resource

### PR DESCRIPTION
**Pull request by chanhsu001**
write target file directly; append .tmp.$PID to the target file as the tmp file

### CODE REVIEW

**Review by quake — CHANGES_REQUESTED** [review]
please check the ci error, the changes of `Cargo.lock` file should be checked-in also.

File: resource/src/lib.rs [hunk 1]
```diff
@@ -271,11 +269,10 @@ mod tests {
     use std::fs;
     use std::path::{Path, PathBuf};
 
-    fn mkdir() -> tempfile::TempDir {
-        tempfile::Builder::new()
-            .prefix("ckb_resource_test")
-            .tempdir()
-            .unwrap()
+    fn mkdir() -> PathBuf {
```
Comment (quake) [inline resource/src/lib.rs:272]: Does `/tmp/ckb_resource_test` folder create successful on windows? I think we may keep the test code unchanged here, just move `tempfile` crate to `[dev-dependencies]` section.

Comment (chanhsu001) [reply resource/src/lib.rs:272]: ok,  you are right, let's move tempfile crate into dev-dependencies and untouch test code.

**Review by quake — APPROVED** [review]

**Review by yangby-cryptape — APPROVED** [review]

**Comment by yangby-cryptape** [comment]
bors merge=quake,yangby-cryptape

**Pull request merged**